### PR TITLE
Switch to pnpm

### PR DESCRIPTION
**Pull request by HerbCaudill**
Replaces yarn 1.x with pnpm as package manager for the repo.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,27 +9,24 @@
...
-    "dev:demo": "yarn workspace @automerge/automerge-repo-demo-todo dev",
-    "dev:svelte-demo": "yarn workspace @automerge/automerge-repo-demo-counter-svelte dev",
+    "build": "pnpm -r build",
+    "dev:demo": "pnpm -F @automerge/automerge-repo-demo-todo dev",
+    "dev:svelte-demo": "pnpm -F @automerge/automerge-repo-demo-counter-svelte dev",
     "dev": "run-p watch start:syncserver dev:demo",
     "pub": "lerna publish --yes",
-    "start:syncserver": "cross-env DEBUG='WebsocketServer' yarn workspace @automerge/example-sync-server start",
+    "start:syncserver": "cross-env DEBUG='WebsocketServer' pnpm -F @automerge/example-sync-server start",
+    "preinstall": "npx only-allow pnpm",
```
Comment (HerbCaudill) [inline package.json:18]: this helps us avoid accidentally running `yarn install` etc

File: packages/automerge-repo-network-messagechannel/package.json [hunk 2]
```diff
@@ -13,7 +13,9 @@
     "test": "vitest"
   },
   "dependencies": {
-    "@automerge/automerge-repo": "^1.1.0-alpha.2"
+    "@automerge/automerge-repo": "workspace:*",
+    "debug": "^4.3.4",
+    "eventemitter3": "^5.0.1"
```
Comment (HerbCaudill) [inline packages/automerge-repo-network-messagechannel/package.json:18]: pnpm is stricter about packages only using dependencies that they declare, whereas yarn let us get away with being sloppy.

**Comment by alexjg** [comment]
This LVGTM. @pvh any concerns?

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-name: Release
+jname: Release
```
Comment (pvh) [inline .github/workflows/release.yaml:1]: jname?

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -29,16 +29,20 @@ jobs:
...
+          run_install: false
       - uses: actions/checkout@v3
         with:
           fetch-depth: 0
           ref: ${{ github.ref }}
       - name: Install and Build
         run: |
-          yarn install --no-lockfile
-          yarn build
+          pnpm install --no-lockfile
```
[the remark this answers is not in the document]
Comment (HerbCaudill) [reply .github/workflows/release.yaml:44]: gah

[the remark this answers is not in the document]
Comment (HerbCaudill) [reply .github/workflows/release.yaml]: Looks like it's the default setting for pnpm in CI. 
https://pnpm.io/cli/install#--frozen-lockfile

**Pull request merged**